cron/esCacheMaint.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+# Copyright (c) 2014 Mozilla Corporation
+
+import json
+import logging
+import pytz
+import random
+import requests
+import sys
+import time
+from configlib import getConfig, OptionParser
+from datetime import datetime, date, timedelta
+from dateutil.parser import parse
+from elasticsearch import Elasticsearch
+logger = logging.getLogger(sys.argv[0])
+
+def loggerTimeStamp(self, record, datefmt=None):
+    return datetime.utcnow().isoformat()
+
+def initLogger():
+    logger.level = logging.INFO
+    formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+    formatter.formatTime = loggerTimeStamp
+    sh = logging.StreamHandler(sys.stderr)
+    sh.setFormatter(formatter)
+    logger.addHandler(sh)
+
+def esConnect(conn):
+    '''open or re-open a connection to elastic search'''
+    return Elasticsearch((options.esservers))
+
+
+def isJVMMemoryHigh():
+    url = "http://{0}/_nodes/stats?pretty=true".format(random.choice(options.esservers))
+    #r=requests.get(url="http://mozdefes5.private.scl3.mozilla.com:9200/_nodes/stats?pretty=true")
+    r = requests.get(url)
+    logger.debug(r)
+    if r.status_code == 200:
+        nodestats=r.json()
+        #logger.debug(json.dumps(nodestats, indent=4, sort_keys=True))
+
+        for node in nodestats['nodes']:
+            loadaverage=nodestats['nodes'][node]['os']['cpu']['load_average']
+            cpuusage=nodestats['nodes'][node]['os']['cpu']['percent']
+            nodename=nodestats['nodes'][node]['name']
+            jvmused=nodestats['nodes'][node]['jvm']['mem']['heap_used_percent']
+            logger.debug('{0}: cpu {1}%  jvm {2}% load average: {3}'.format(nodename,cpuusage,jvmused,loadaverage))
+            if jvmused> options.jvmlimit:
+                logger.info('{0}: cpu {1}%  jvm {2}% load average: {3} recommending cache clear'.format(nodename,cpuusage,jvmused,loadaverage))
+                return True
+        return False
+    else:
+        logger.error(r)
+        return False
+
+def clearESCache():
+    es=esConnect(None)
+    indexes=es.indices.stats()['indices']
```

Comment:
It's preferred to use our elasticsearch client to get this list https://github.com/mozilla/MozDef/blob/master/lib/elasticsearch_client.py#L48